include/dali/core/tensor_shape.h

```diff
@@ -793,19 +832,19 @@ struct TensorListShape : TensorListShapeBase<TensorListShape<sample_ndim>, sampl
       : Base(flatten_shapes(sample_shapes), sample_shapes.size()) {}
 
   TensorListShape(const std::vector<int64_t> &shapes, int sample_dim)
-      : Base(shapes, shapes.size() / sample_dim) {
+      : Base(shapes, shapes.size() / (sample_dim == 0 ? 1 : sample_dim)) {
     assert(sample_dim == sample_ndim);
   }
 
   TensorListShape(std::vector<int64_t> &&shapes, int sample_dim)
-      : Base(std::move(shapes), shapes.size() / sample_dim) {
+      : Base(std::move(shapes), shapes.size() / (sample_dim == 0 ? 1 : sample_dim)) {
     assert(sample_dim == sample_ndim);
   }
 
   TensorListShape(const std::vector<int64_t> &shapes, int num_samples, int sample_dim)
       : Base(shapes, num_samples) {
     assert(sample_dim == sample_ndim);
-    assert(num_samples == static_cast<int>(shapes.size()) / sample_dim);
+    assert(num_samples == static_cast<int>(shapes.size()) / (sample_dim == 0 ? 1 : sample_dim));
```

Comment:
Bug
```suggestion
    assert(num_samples * sample_dim == static_cast<int>(shapes.size()));
```

Reply:
I believe, that it's incorrect:
```c++
ndims=0
num_samples=4
shapes: {1, 1, 1, 1}

num_samples * sample_dim == 0
static_cast<int>(shapes.size()) == 4
```
Unless `shapes` should look differently in this case?